lib/consult.rb

```diff
@@ -24,7 +24,15 @@ def load(config_dir: nil, force_render: false, verbose: nil)
       root directory: config_dir
       yaml = root.join('config', 'consult.yml')
 
-      @all_config = yaml.exist? ? YAML.safe_load(ERB.new(yaml.read).result, [], [], true, symbolize_names: true).to_h : {}
+      @all_config = if yaml.exist?
+        if Gem::Version.new(YAML::VERSION) < Gem::Version.new('4.0')
+          YAML.safe_load(ERB.new(yaml.read).result, [], [], true, symbolize_names: true).to_h
```

Comment:
Non-blocking comments:
In this call signature what are the 2nd and 3rd args (`[]`)? `legacy_permitted_classes` & `legacy_permitted_symbols`?

Is the 4th arg (`true) for `aliases` and can it be a named arg for greater clarity?

My curiosity was triggered by reading [the source for Psych 3.3.0](https://github.com/ruby/psych/blob/0abce07b908fa0128223a3e11f2eeb4be0c191d6/lib/psych.rb#L329) which is what Ruby 2.7.3 is using in my local.

Reply:
> In this call signature what are the 2nd and 3rd args ([])? legacy_permitted_classes & legacy_permitted_symbols?

I assume so 🤷‍♂️

> Is the 4th arg (true) for `aliases` and can it be a named arg for greater clarity?

Again, I assume so; the objective of this code was to make as close to a 100% safe change as possible for the existing code path, but then fix it in the future code path. At some point we can come back here and delete the old code (probably when we drop support for Ruby < 3.1)